x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/EnrichPolicyResolver.java

```diff
@@ -101,7 +106,16 @@ public EnrichPolicyResolver(
     }
 
     public record UnresolvedPolicy(String name, Enrich.Mode mode) {
+        public static UnresolvedPolicy from(Enrich e) {
+            return new UnresolvedPolicy(asString(e.policyName()), e.mode());
+        }
 
+        private static String asString(Expression expression) {
```

Comment:
In this PR, Julian is adding this function to the Foldables class: https://github.com/elastic/elasticsearch/pull/131870/files#diff-dbc8ae6bfd90922dda68e61b020be920e8514b5870a999e8c85923f7e3f6d3a7R20-R25

I would move this there, to have a conflict later and avoid repeated functionality. As the error string is different, maybe having an overload accepting the error?